### PR TITLE
Add interface level object definitions for BeaconBlock endpoint.

### PR DESCRIPTION
**Pull request by rolfyone**
Added tests for some utility functions, moved functional tests from the interface level down to the ChainDataProvider.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

#1301

### CODE REVIEW

File: data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/CacheControlUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+@ExtendWith(MockitoExtension.class)
+public class CacheControlUtilsTest {
+
+  SignedBeaconBlock signedBlock =
+      new SignedBeaconBlock(DataStructureUtil.randomSignedBeaconBlock(1, 1));
+  @Mock private ChainDataProvider provider;
+
+  @Test
+  void getMaxAgeForSignedBlock_shouldSetZeroIfNotFinalized() {
```
Comment (macfarla) [inline data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/CacheControlUtilsTest.java]:
```suggestion
  void getMaxAgeForSignedBlock_shouldReturnZeroIfNotFinalized() {
```

File: data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/CacheControlUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    when(provider.isFinalized(signedBlock)).thenReturn(false);
+    String cacheControl = CacheControlUtils.getMaxAgeForSignedBlock(provider, signedBlock);
+    assertThat(cacheControl).isEqualTo(CACHE_NONE);
+  }
+
+  @Test
+  void getMaxAgeForSignedBlock_shouldSetZeroIfFinalized() {
```
Comment (macfarla) [inline data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/CacheControlUtilsTest.java]:
```suggestion
  void getMaxAgeForSignedBlock_shouldReturnZeroIfFinalized() {
```

Comment (macfarla) [reply data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/CacheControlUtilsTest.java]: hang on they aren't both zero are they?

File: data/provider/src/main/java/tech/pegasys/artemis/api/schema/BLSPubKey.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+public class BLSPubKey {
+  /** The number of bytes in this value - i.e. 48 */
+  private final int SIZE = 48;
+
+  private final Bytes bytes;
+
+  public BLSPubKey(Bytes bytes) {
+    checkArgument(
+        bytes.size() == SIZE, "Bytes4 should be %s bytes, but was %s bytes.", SIZE, bytes.size());
```
Comment (macfarla) [inline data/provider/src/main/java/tech/pegasys/artemis/api/schema/BLSPubKey.java]: why is Bytes4 48 bytes?

File: data/provider/src/main/java/tech/pegasys/artemis/api/schema/BLSSignature.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+public class BLSSignature {
+  /** The number of bytes in this value - i.e. 96 */
+  private final int SIZE = 96;
+
+  private final Bytes bytes;
+
+  public BLSSignature(Bytes bytes) {
+    checkArgument(
+        bytes.size() == SIZE, "Bytes4 should be %s bytes, but was %s bytes.", SIZE, bytes.size());
```
Comment (macfarla) [inline data/provider/src/main/java/tech/pegasys/artemis/api/schema/BLSSignature.java]: should it just be Bytes should be ...

File: data/provider/src/main/java/tech/pegasys/artemis/provider/BLSSignatureSerializer.java [hunk 5]
```diff
@@ -17,12 +17,12 @@
...
 import java.io.IOException;
-import tech.pegasys.artemis.util.bls.BLSSignature;
+import tech.pegasys.artemis.api.schema.BLSSignature;
 
 public class BLSSignatureSerializer extends JsonSerializer<BLSSignature> {
   @Override
   public void serialize(BLSSignature value, JsonGenerator gen, SerializerProvider serializers)
       throws IOException {
-    gen.writeString(value.toBytes().toHexString().toLowerCase());
+    gen.writeString(value.toHexString().toLowerCase());
```
Comment (macfarla) [inline data/provider/src/main/java/tech/pegasys/artemis/provider/BLSSignatureSerializer.java:26]: is this change related?

File: data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/CacheControlUtilsTest.java (hunk 2 above)
Comment (rolfyone) [reply data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/CacheControlUtilsTest.java]: ok, bad naming of the test.
If the object is not finalized, i want to set max-age=zero in the cache header, otherwise setting to the cache-finalized string (cache timeout = 1 year)
Hopefully the new names make more sense..

File: data/provider/src/main/java/tech/pegasys/artemis/provider/JsonProvider.java [hunk 6]
```diff
@@ -38,6 +39,7 @@ private void addTekuMappers() {
 
     module.addDeserializer(BLSPublicKey.class, new BLSPublicKeyDeserializer());
     module.addSerializer(BLSPublicKey.class, new BLSPublicKeySerializer());
+    module.addSerializer(BLSPubKey.class, new BLSPubKeySerializer());
```
Comment (macfarla) [inline data/provider/src/main/java/tech/pegasys/artemis/provider/JsonProvider.java:41]: why do we have BLSPublicKey and BLSPubKey?

File: data/provider/src/test/java/tech/pegasys/artemis/api/ChainDataProviderTest.java [hunk 7]
```diff
@@ -51,9 +53,9 @@
   private static UnsignedLong slot;
   private static EventBus localEventBus;
   private static ChainStorageClient chainStorageClient;
-
-  @SuppressWarnings("unused")
-  private static UnsignedLong epoch;
+  private final tech.pegasys.artemis.datastructures.blocks.SignedBeaconBlock signedBeaconBlock =
+      DataStructureUtil.randomSignedBeaconBlock(1, 1);
+  //  private static UnsignedLong epoch;
```
Comment (macfarla) [inline data/provider/src/test/java/tech/pegasys/artemis/api/ChainDataProviderTest.java]: stray comment - is epoch not needed?

File: data/provider/src/test/java/tech/pegasys/artemis/api/ChainDataProviderTest.java [hunk 8]
```diff
@@ -174,4 +176,86 @@ public void getGenesisTime_shouldReturnValueIfStoreAvailable() {
...
+    SignedBeaconBlock result = future.get().get();
+    assertThat(result)
+        .usingRecursiveComparison()
+        .isEqualTo(new SignedBeaconBlock(signedBeaconBlock));
+  }
+
+  @Test
+  public void getBlockByBlockRoot_shouldReturnEmptyWhenStoreNotFound()
+      throws ExecutionException, InterruptedException {
+    CombinedChainDataClient myClient = mock(CombinedChainDataClient.class);
```
Comment (macfarla) [inline data/provider/src/test/java/tech/pegasys/artemis/api/ChainDataProviderTest.java]: myClient??

File: data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconBlockHandler.java [hunk 9]
```diff
@@ -66,65 +77,61 @@ public BeaconBlockHandler(
...
+      } else if (queryParamMap.size() > 1) {
+        throw new IllegalArgumentException(TOO_MANY_PARAMETERS);
       }
 
-      final Map<String, List<String>> queryParamMap = ctx.queryParamMap();
-      if (ctx.queryParamMap().containsKey(ROOT)) {
-        final Bytes32 blockParam =
-            Bytes32.fromHexString(validateQueryParameter(queryParamMap, ROOT));
+      if (queryParamMap.containsKey(ROOT)) {
+        final Bytes32 blockParam = getParameterValueAsBytes32(queryParamMap, ROOT);
```
Comment (macfarla) [inline data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconBlockHandler.java]:
```suggestion
        final Bytes32 blockRoot = getParameterValueAsBytes32(queryParamMap, ROOT);
```

**Review by macfarla — COMMENTED** [review]
looks good overall. some comments

File: data/provider/src/main/java/tech/pegasys/artemis/api/schema/BLSPubKey.java (hunk 3 above)
Comment (rolfyone) [reply data/provider/src/main/java/tech/pegasys/artemis/api/schema/BLSPubKey.java]: comment should be bringing in size, i'll fix those.

File: data/provider/src/main/java/tech/pegasys/artemis/provider/BLSSignatureSerializer.java (hunk 5 above)
Comment (rolfyone) [reply data/provider/src/main/java/tech/pegasys/artemis/provider/BLSSignatureSerializer.java:26]: because BLSSignature changed, and the toBytes isn't on the new object... We're the only consumers of the serializers, so i updated it.

File: data/provider/src/main/java/tech/pegasys/artemis/provider/JsonProvider.java (hunk 6 above)
Comment (rolfyone) [reply data/provider/src/main/java/tech/pegasys/artemis/provider/JsonProvider.java:41]: as discussed, will stick with spec name, and remove the serializer for BLSPublicKey

File: data/provider/src/test/java/tech/pegasys/artemis/api/ChainDataProviderTest.java (hunk 8 above)
Comment (rolfyone) [reply data/provider/src/test/java/tech/pegasys/artemis/api/ChainDataProviderTest.java]: renamed

File: data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconBlockHandler.java (hunk 9 above)
Comment (rolfyone) [reply data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconBlockHandler.java]: will do this as part of bigger commit.

**Review by macfarla — APPROVED** [review]
